lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -264,81 +267,112 @@ export const SignInUpFeature: React.FC<
+
+                await setLoginAttemptInfo({
+                    recipeImplementation: recipe.recipeImpl,
+                    userContext: input.userContext,
+                    attemptInfo: loginAttemptInfo,
+                });
                 dispatch({ type: "startLogin", loginAttemptInfo });
-            } else if (res.status === "GENERAL_ERROR") {
-                dispatch({ type: "setError", error: res.message });
             }
             return res;
         },
         resendCode: async (input) => {
-            const res = await originalImpl.resendCode(input);
-            if (res.status === "OK") {
-                const loginAttemptInfo = await originalImpl.getLoginAttemptInfo();
-                // If it was cleared or overwritten we don't want to save this.
-                // TODO: extend session checker to check for this case as well
-                if (loginAttemptInfo !== undefined && loginAttemptInfo.deviceId === input.deviceId) {
-                    const timestamp = new Date().getTime();
-                    await originalImpl.setLoginAttemptInfo({ ...loginAttemptInfo, lastResend: timestamp });
-                    dispatch({ type: "resendCode", timestamp });
+            try {
+                const res = await recipe.recipeImpl.resendCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = await getLoginAttemptInfo({
+                        recipeImplementation: recipe.recipeImpl,
+                        userContext: input.userContext,
+                    });
+                    // If it was cleared or overwritten we don't want to save this.
+                    // TODO: extend session checker to check for this case as well
+                    if (loginAttemptInfo !== undefined && loginAttemptInfo.deviceId === input.deviceId) {
+                        const timestamp = Date.now();
+
+                        await setLoginAttemptInfo({
+                            recipeImplementation: recipe.recipeImpl,
+                            userContext: input.userContext,
+                            attemptInfo: {
+                                ...loginAttemptInfo,
+                                lastResend: timestamp,
+                            },
+                        });
+                        dispatch({ type: "resendCode", timestamp });
+                    }
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await recipe.recipeImpl.clearLoginAttemptInfo({
+                        userContext: input.userContext,
+                    });
+
+                    dispatch({ type: "restartFlow", error: "ERROR_SIGN_IN_UP_RESEND_RESTART_FLOW" });
+                }
+                return res;
+            } catch (e) {
+                if (STGeneralError.isThisError(e)) {
+                    dispatch({ type: "setError", error: e.message });
                 }
```

Comment:
How come this is not being done in the other functions in this file (like `consumeCode`)